### PR TITLE
Implemented #667 related jobs (once part of #137)

### PR DESCRIPTION
**Pull request by TimZaman**
See title.
related_jobs

### CODE REVIEW

File: digits/job.py [hunk 1]
```diff
@@ -266,3 +266,10 @@ def is_read_only(self):
         Returns False if this job can be edited
         """
         return False
+
+    def parent_jobs(self):
+        """
+        Returns the list of the jobs on which the job
+        depends
+        """
+        raise NotImplementedError('Implement me!') #@TODO
```
Comment (gheinrich) [inline digits/job.py]: you could return an empty list to avoid having to catch the exception in the caller?

File: digits/scheduler.py [hunk 2]
```diff
@@ -208,6 +208,20 @@ def get_job(self, job_id):
             return None
         return self.jobs.get(job_id, None)
 
+    def get_related_jobs(self, job):
+        """
+        Look through self.jobs to try to find the Jobs
+        whose parent contains job
+        """
+        related_jobs = []
```
Comment (gheinrich) [inline digits/scheduler.py]: I think this doesn't return anything for model jobs. Should you initialize as `related_jobs = job.parents_job()` to capture the dataset associated with model jobs?

**Comment by gheinrich** [comment]
Thanks for submitting this PR! Can you define what is meant by "related" job?

**Comment by TimZaman** [comment]
In the dataset view, it shows you the associated models that ran on the dataset..

**Comment by gheinrich** [comment]
You made changes in the model view too though there doesn't seem to be a way to show any related jobs on those? I thought you would show models that use the same dataset but that isn't how the code implements this. Would you like to revert your changes in the model files?

Also, if you haven't done so already, would you mind signing the [CLA](https://github.com/gheinrich/DIGITS/blob/master/README.md#bugs-and-feature-requests). Thanks!

File: digits/job.py (hunk 1 above)
Comment (TimZaman) [inline digits/job.py]: I was just following the repository's convention. But i agree the exceptions are quite dirty, will change.
https://github.com/NVIDIA/DIGITS/blob/master/digits/job.py#L152-L157

File: digits/scheduler.py (hunk 2 above)
Comment (TimZaman) [inline digits/scheduler.py]: Correct. Will implement that too.

**Comment by TimZaman** [comment]
I have now made so many improvements upon digits that this one's already outdated. Gonna close it this pull request. Might open another one if i get to it.

**Pull request closed without merging**